src/common/dout_fmt.h

```diff
@@ -0,0 +1,56 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright contributors to the Ceph project
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#pragma once
+
+#include <iosfwd>
+#include <iterator>
+#include <fmt/core.h>
+#include "dout.h"
+
+/// \file dout_fmt.h
+///
+/// \brief dout macros to format log statements with libfmt
+///
+/// A set of dout macros taking a format string and its corresponding argument
+/// list. Log output is written directly to the underlying std::ostream by
+/// fmt::format_to() rather than exposing the stream for ostream operator
+/// chaining.
+
+// work around "warning: value computed is not used" with default dout_prefix
+inline void dout_fmt_use_prefix(std::ostream&) {}
+
+#define lsubdout_fmt(cct, sub, v, ...) \
+  dout_impl(cct, ceph_subsys_##sub, v) \
+  dout_fmt_use_prefix(dout_prefix); \
+  fmt::format_to(std::ostream_iterator<char>(*_dout), __VA_ARGS__); \
```

Comment:
```suggestion
  fmt::print(*_dout, __VA_ARGS__); \
```

IIRC, `fmt::print()` is able to print to an `ostream`, see https://fmt.dev/latest/api.html#_CPPv4IDpEN3fmt5printEvRNSt7ostreamE13format_stringIDp1TEDpRR1T

can we use it directly?

Reply:
would it be more or less efficient?

Reply:
i propose this just because it's simpler. i don't think performance-wise it is different from what Casey's implementatino.

Reply:
ok.  if casey had written it with print, I wouldn't have noticed, just wondered

Reply:
thanks @tchaikov! i wasn't aware of `fmt::print()`

> would it be more or less efficient?

the ostream version is potentially more efficient since ostream_iterator can only write one output byte at a time while ostream::write() can write several